kornia/geometry/transform/thin_plate_spline.py

```diff
@@ -0,0 +1,220 @@
+    t1_sq: torch.Tensor = tensor1.mul(tensor1).sum(dim=-1, keepdim=True)
+    t2_sq: torch.Tensor = tensor2.mul(tensor2).sum(dim=-1, keepdim=True).transpose(1, 2)
+    t1_t2: torch.Tensor = tensor1.matmul(tensor2.transpose(1, 2))
+    square_dist: torch.Tensor = -2 * t1_t2 + t1_sq + t2_sq
+    square_dist: torch.Tensor = square_dist.clamp(min=0)  # handle possible numerical errors
+    return square_dist
+
+
+def _kernel_distance(squared_distances: torch.Tensor, eps: float = 1e-8) -> torch.Tensor:
+    r"""Compute the TPS kernel distance function: :math:`r^2 log(r)`, where `r` is the euclidean distance.
+    Since :math:`\log(r) = 1/2 \log(r^2)`, this function takes the squared distance matrix and calculates
+    :math:`0.5 r^2 log(r^2)`."""
+    # r^2 * log(r) = 1/2 * r^2 * log(r^2)
+    return 0.5 * squared_distances * squared_distances.add(eps).log()
+
+
+def get_tps_transform(points_src: torch.Tensor, points_dst: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+    r"""Compute the TPS transform parameters that warp source points to target points.
+
+    The input to this function is a tensor of :math:`(x, y)` source points :math:`(B, N, 2)` and a corresponding
+    tensor of target :math:`(x, y)` points :math:`(B, N, 2)`.
+
+    Args:
+        points_src (torch.Tensor): batch of source points :math:`(B, N, 2)` as :math:`(x, y)` coordinate vectors.
+        points_dst (torch.Tensor): batch of target points :math:`(B, N, 2)` as :math:`(x, y)` coordinate vectors.
+
+    Returns:
+        Tuple[torch.Tensor, torch.Tensor]: :math:`(B, N, 2)` tensor of kernel weights and :math:`(B, 3, 2)`
+            tensor of affine weights. The last dimension contains the x-transform and y-transform weights
+            as seperate columns.
+
+    Example:
+        >>> points_src = torch.rand(1, 5, 2)
+        >>> points_dst = torch.rand(1, 5, 2)
+        >>> kernel_weights, affine_weights = get_tps_transform(points_src, points_dst)
+    """
+    if not isinstance(points_src, torch.Tensor):
+        raise TypeError(f"Input points_src is not torch.Tensor. Got {type(points_src)}")
+
+    if not isinstance(points_dst, torch.Tensor):
+        raise TypeError(f"Input points_dst is not torch.Tensor. Got {type(points_dst)}")
+
+    if not len(points_src.shape) == 3:
+        raise ValueError(f"Invalid shape for points_src, expected BxNx2. Got {points_src.shape}")
+
+    if not len(points_dst.shape) == 3:
+        raise ValueError(f"Invalid shape for points_dst, expected BxNx2. Got {points_dst.shape}")
+
+    device, dtype = _extract_device_dtype([points_src, points_dst])
+    batch_size, num_points = points_src.shape[:2]
+
+    # set up and solve linear system
+    # [K   P] [w] = [dst]
+    # [P^T 0] [a]   [ 0 ]
+    pair_distance: torch.Tensor = _pair_square_euclidean(points_src, points_dst)
+    k_matrix: torch.Tensor = _kernel_distance(pair_distance)
+
+    requires_grad = points_src.requires_grad or points_dst.requires_grad
```

Comment:
where is this used ?

Reply:
I'm not sure how that got there, maybe an accidental copy/paste...